src/main/java/uk/gov/pay/webhooks/webhook/resource/WebhookResource.java

```diff
@@ -166,17 +172,19 @@ public List<WebhookResponse> getWebhooks(@Parameter(example = "true", descriptio
                                              @NotNull @QueryParam("live") Boolean live,
                                              @Parameter(example = "eo29upsdkjlk3jpwjj2dfn12", description = "Service external ID. Required when override_service_id_restriction is not `true`")
                                              @QueryParam("service_id") String service_id,
+                                             @Parameter(example = "100", description = "Gateway account ID. Required when override_service_id_restriction is not `true`")
+                                             @QueryParam("gateway_account_id") String gatewayAccountId,
                                              @Parameter(example = "true", description = "Set to true to list all webhooks. if 'true', service_id is not permitted")
                                              @QueryParam("override_service_id_restriction") boolean overrideServiceIdRestriction) {
-        if (service_id != null && overrideServiceIdRestriction) {
-            throw new BadRequestException("service_id not permitted when using override_service_id_restriction");
+        if ((service_id != null || gatewayAccountId != null) && overrideServiceIdRestriction) {
+            throw new BadRequestException("[service_id, gateway_account_id] not permitted when using override_service_id_restriction");
         }
 
-        if (service_id == null && !overrideServiceIdRestriction) {
-            throw new BadRequestException("either service_id or override_service_id_restriction query parameter must be provided");
+        if ((service_id == null || gatewayAccountId == null) && !overrideServiceIdRestriction) {
```

Comment:
Are there some tests for the various logic for this on different endpoints that we can add to to ensure that gateway_account_id is required as expected?